run.py

```diff
@@ -680,29 +691,63 @@ def main():
   # Create output directory
   output_dir = create_output(args.o, args.noclean)
 
-  # Run directed assembly tests
+  # Run any handcoded/directed assembly tests specified by args.asm_tests
   if args.asm_tests != "":
     asm_test = args.asm_tests.split(',')
     for path_asm_test in asm_test:
-      if os.path.isdir(path_asm_test):    # Path asm test is a directory
-        run_assembly_from_dir(path_asm_test, args.iss_yaml, args.isa, args.mabi,
-                              args.iss, output_dir, args.core_setting_dir)
-      else:                               # Path asm test is a assembly file
-        run_assembly(path_asm_test, args.iss_yaml, args.isa, args.mabi, args.iss,
-                     output_dir, args.core_setting_dir)
+      full_path = os.path.expanduser(path_asm_test)
+      # path_asm_test is a directory
+      if os.path.isdir(full_path):
+        run_assembly_from_dir(full_path, args.iss_yaml, args.isa, args.mabi,
+                              args.gcc_opts, args.iss, output_dir, args.core_setting_dir)
+      # path_asm_test is an assembly file
+      elif os.path.isfile(full_path):
+        run_assembly(full_path, args.iss_yaml, args.isa, args.mabi, args.gcc_opts,
+                     args.iss, output_dir, args.core_setting_dir)
+      else:
+        logging.error('%s does not exist' % full_path)
     return
 
   subprocess.run(["mkdir", "-p", ("%s/asm_tests" % output_dir)])
   # Process regression test list
   matched_list = []
+  # Any tests in the YAML test list that specify a directed assembly test
+  directed_list = []
 
   if not args.co:
     process_regression_list(args.testlist, args.test, args.iterations, matched_list, cwd)
-    if len(matched_list) == 0:
-      sys.exit("Cannot find %s in %s" % (args.test, args.testlist))
+    for t in list(matched_list):
+      if 'asm_tests' in t:
```

Comment:
I think we should create a different entry name for a directed assembly test instead of using the name `test:` field in yaml. It may be name `direct_test:`. The `direct_test` does not need the field like `sim_opts, gen_opts,...`
User will not be confused between gen test, and direct assembly test. What do you think?

Reply:
I don't think this would be necessary, as in the example that I added to the README in this PR, you don't need to include the `sim_opts, gen_opts, ...` to any directed assembly test entries anyways. To make this more clear, I will specify in the README that for any directed test entries that use `asm_tests` field the `gen_test` field must not be defined simultaneously.